Dockerfile

```diff
@@ -0,0 +1,29 @@
+# Copyright 2021 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+FROM golang:1.16 AS build-env
+RUN mkdir -p /go/src/sig.k8s.io/gateway-api
+WORKDIR /go/src/sig.k8s.io/gateway-api
+COPY  . .
+RUN useradd -u 10001 webhook
+ARG TAG
+ARG COMMIT
+RUN CGO_ENABLED=0 GOARCH=amd64 GOOS=linux go build -a -o gateway-api-webhook \
+      -ldflags "-s -w -X main.VERSION=$TAG -X main.COMMIT=$COMMIT" ./cmd/admission
+
+FROM scratch
```

Comment:
drive by comment: maybe distroless is a better choice? Or whatever base image core Kubernetes is standardizing on?

Reply:
Thanks for the nudge. I switched to `gcr.io/distroless/static:nonroot` instead which simplified the Dockerfile a bit too. PTAL.

Reply:
I've borrowed the base image from Cluster API:https://github.com/kubernetes-sigs/cluster-api/blob/master/Dockerfile#L56